src/dmatrix.jl

```diff
@@ -435,15 +485,21 @@ _numpy_json_typestr(::Type{<:Complex{<:AbstractFloat}}) = "c"
 
 numpy_json_typestr(::Type{T}) where {T<:Number} = string("<",_numpy_json_typestr(T),sizeof(T))
 
-function numpy_json_info(x::AbstractMatrix; read_only::Bool=false)
-    info = Dict("data"=>(convert(Csize_t, pointer(x)), read_only),
-                "shape"=>reverse(size(x)),
-                "typestr"=>numpy_json_typestr(eltype(x)),
-                "version"=>3,
-               )
-    JSON3.write(info)
+# yes, this is supposed to also work for CuArray
```

Comment:
This comment is unclear. Could the comment maybe be rewritten in such a way that it clearly answers a "why?" question?

Reply:
I can definitely  see how this comment would be confusing.  I have updated it to give more context.